experiments/basic_skeleton.ts

```diff
@@ -0,0 +1,107 @@
+import { url_class } from '../src/url_class';
+
+// Given a Repo URL, uses REGEX to grab the repo and owner name
+// I **THINK** it works with any (GitHub) URL they can throw at us, but I haven't exhaustively tested it
+// To test the REGEX function, compile and run regextext.ts with different inputs.
+
+function getGitRepoDetails(url: string): {username: string, repoName: string} | null {
```

Comment:
I believe we could use a library for this instead. to reduce the amount of code we're writing. 

I think Koltan mentioned https://github.com/jonschlinkert/parse-github-url as a possible library for this. Unless you are very confident that this regex always works.

Reply:
I installed it and it had 4 security vulnerabilities, so I decided to do it manually. Works for everything I've tested so far

Reply:
Ok good call